src/sentry/eventstream/kafka/protocol.py

```diff
@@ -129,6 +136,20 @@ def decode_bool(value: bytes) -> bool:
     return bool(int(decode_str(value)))
 
 
+def decode_optional_dict_str(value: Optional[bytes]) -> Optional[Mapping[Any, Any]]:
+    if value is None:
+        return None
+    dict_str = decode_optional_str(value)
+    if dict_str is None:
+        return None
+
+    parsed = json.loads(dict_str)
```

Comment:
So se the comment above. One of the reasons we used headers was to avoid decoding json in the post process forwarder. Now this is likely to be mostly small but we should do it only if we really need.